core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -1447,8 +1451,9 @@ private[spark] object MapOutputTracker extends Logging {
     // TODO: SPARK-35036: Instead of reading map blocks in case of AQE with Push based shuffle,
     // TODO: improve push based shuffle to read partial merged blocks satisfying the start/end
     // TODO: map indexes
-    if (mergeStatuses.exists(_.nonEmpty) && startMapIndex == 0
-      && endMapIndex == mapStatuses.length) {
+    if (mergeStatuses.exists(_.nonEmpty) && mergeStatuses.exists(_.exists(_ != null))
+      && startMapIndex == 0 && endMapIndex == mapStatuses.length) {
+      enableBatchFetch = false
```

Comment:
Would it make sense to add a log line here? Would be useful for debugging. If not `INFO` atleast `DEBUG`